scripts/update-index.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# This script update 'docs/index.md' with 'README.md' 
+#  changing github/learn-ocaml/docs URLs to relative path.
+
+readme="$PWD/README.md"
+index="$PWD/docs/index.md"
+
+if [[ ! -r "$readme" ]]
+then
+  echo "Error: '$readme': file does not exist or unreadable"
+  exit 1 
+fi
+
+if [[ ! -w "$index" ]]
+then
+  echo "Error: '$index': file does not exist or unwritable"
+  exit 1 
+fi
+
+buffin=$(<$readme)
+buffout=""
+
+url="https://github.com/ocaml-sf/learn-ocaml/blob/master/docs"
+
+while IFS= read -r line
+do
+
+  # Using sed command with "; semicolon" separator to avoid any problems 
+  #  with the "/ slash" contained in URLs.
+  line=$(echo "$line" | sed -r -e 's;'"$url"';.;g')
+
+  # The read function delete the newline, we need to re-inset it.
+  buffout="$buffout$line"$'\n'
+done <<< "$buffin"
+
+# Writing index, deleting the last unnecessary new line
+#  inserted during the process above.
+echo "${buffout::-1}" > $index
```

Comment:
Sub-optimal echo / `::-1`, and missing double-quote:

```suggestion
printf '%s' "$buffout" > "$index"
```

Indeed, `echo` was adding an extraneous `\n`, while `printf '%s' …` prints the text as is (which already contains the necessary `\n`)

Reply:
I did not know I can use `printf` instead of `echo` but it seems to be a better way to do.